From baffe9948ceebb3881867eb5a3ad34b1e725478b Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Mon, 21 Jul 2025 10:52:56 +0800
Subject: [PATCH 222/236] mtk: hostapd: use assoc_sm's auth_alg when validating
 PMKSA

When processing partner links, the sm->auth_alg is not set here and
therefore is not usable. Instead assoc_sm->auth_alg should be use.

The sm->auth_alg is set later in ieee80211_ml_process_link().

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 src/ap/wpa_auth_ie.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/ap/wpa_auth_ie.c b/src/ap/wpa_auth_ie.c
index 83ae4e065..62152c089 100644
--- a/src/ap/wpa_auth_ie.c
+++ b/src/ap/wpa_auth_ie.c
@@ -1291,6 +1291,7 @@ wpa_validate_wpa_ie(struct wpa_authenticator *wpa_auth,
 		u64 drv_flags = 0;
 		u64 drv_flags2 = 0;
 		bool ap_sae_offload = false;
+		u16 auth_alg = assoc_sm ? assoc_sm->auth_alg : sm->auth_alg;
 
 		if (wpa_auth->cb->get_drv_flags &&
 		    wpa_auth->cb->get_drv_flags(wpa_auth->cb_ctx, &drv_flags,
@@ -1313,7 +1314,7 @@ wpa_validate_wpa_ie(struct wpa_authenticator *wpa_auth,
 		 * can be used regardless of which PMKID(s) are indicated in the
 		 * (Re)Association Request frame. */
 		if (!ap_sae_offload && data.num_pmkid && !sm->pmksa &&
-		    sm->auth_alg == WLAN_AUTH_OPEN) {
+		    auth_alg == WLAN_AUTH_OPEN) {
 			wpa_auth_vlogger(wpa_auth, sm->addr, LOGGER_DEBUG,
 					 "No PMKSA cache entry found for SAE");
 			return WPA_INVALID_PMKID;
-- 
2.45.2

